core/services/ocr2/plugins/ocr2keeper/util.go

```diff
@@ -122,31 +122,33 @@ func EVMDependencies21(spec job.Job, db *sqlx.DB, lggr logger.Logger, set evm.Ch
 	// get the chain from the config
 	chainID, err2 := spec.OCR2OracleSpec.RelayConfig.EVMChainID()
 	if err2 != nil {
-		return nil, nil, nil, nil, nil, err2
+		return nil, nil, nil, nil, nil, nil, err2
 	}
 	chain, err2 = set.Get(big.NewInt(chainID))
 	if err2 != nil {
-		return nil, nil, nil, nil, nil, fmt.Errorf("%w: %s", ErrNoChainFromSpec, err2)
+		return nil, nil, nil, nil, nil, nil, fmt.Errorf("%w: %s", ErrNoChainFromSpec, err2)
 	}
 
 	// the provider will be returned as a dependency
 	if keeperProvider, err = EVMProvider(db, chain, lggr, spec, pr); err != nil {
-		return nil, nil, nil, nil, nil, err
+		return nil, nil, nil, nil, nil, nil, err
 	}
 
 	rAddr := ethkey.MustEIP55Address(oSpec.ContractID).Address()
 	if registry, err = kevm21.NewEVMRegistryService(rAddr, chain, mc, lggr); err != nil {
-		return nil, nil, nil, nil, nil, err
+		return nil, nil, nil, nil, nil, nil, err
 	}
 	encoder := kevm21.EVMAutomationEncoder21{}
 
+	hp := kevm21.NewBlockSubscriber(chain.HeadBroadcaster(), chain.LogPoller(), 128, lggr)
+
 	// lookback blocks is hard coded and should provide ample time for logs
 	// to be detected in most cases
 	var lookbackBlocks int64 = 250
 	// TODO: accept a version of the registry contract and use the correct interfaces
 	logTransmitter, err := kevm21.NewTransmitEventProvider(lggr, chain.LogPoller(), rAddr, chain.Client(), lookbackBlocks)
 
-	return keeperProvider, registry, encoder, logTransmitter, registry.LogEventProvider(), err
+	return keeperProvider, registry, encoder, logTransmitter, registry.LogEventProvider(), hp, err
```

Comment:
At some point we should make this a struct instead of a bunch of returned items. lol